sonar-java-plugin/src/main/java/org/sonar/plugins/java/JavaSensor.java

```diff
@@ -184,10 +185,13 @@ private Iterable<InputFile> javaFiles(InputFile.Type type) {
   }
 
   private JavaVersion getJavaVersion() {
-    JavaVersion javaVersion = settings.get(JavaVersion.SOURCE_VERSION)
-      .map(JavaVersionImpl::fromString)
-      .orElse(new JavaVersionImpl());
-    LOG.info("Configured Java source version (" + JavaVersion.SOURCE_VERSION + "): " + javaVersion);
+    Map<String, String> javaVersionProperties = Map.of(
+      JavaVersion.ENABLE_PREVIEW, settings.get(JavaVersion.ENABLE_PREVIEW).orElseGet(() -> ""),
+      JavaVersion.SOURCE_VERSION, settings.get(JavaVersion.SOURCE_VERSION).orElseGet(() -> ""));
+    
+    JavaVersion javaVersion = JavaVersionImpl.fromMap(javaVersionProperties);
```

Comment:
Not sure this is the best way... For me, a constructor with both parameters doing the validation would be fine.